src/cmakeProject.ts

```diff
@@ -2794,21 +2798,49 @@ export class CMakeProject {
         }
 
         const mainListFile = path.join(this.sourceDir, 'CMakeLists.txt');
+        const mainPresetsFile = path.join(this.sourceDir, 'CMakePresets.json');
 
-        if (await fs.exists(mainListFile)) {
-            void vscode.window.showErrorMessage(localize('cmakelists.already.configured', 'A CMakeLists.txt is already configured!'));
-            return -1;
+        if (!await fs.exists(mainListFile)) {
+            if (await this.createCMakeListsFile(mainListFile) !== 0) {
+                return -1;
+            }
+            log.info(localize('cmakelists.created', '[Quick Start] CmakeLists.txt created successfully.'));
+        } else {
+            log.info(localize('cmakelists.exists', '[Quick Start] A CmakeLists.txt already exists, continuing...'));
         }
 
+        // By now, we have a valid CMakeLists.txt so move onto CMakePresets.json
+        if (await fs.exists(mainPresetsFile)) {
+            log.info(localize('cmakepresets.already.configured', '[Quick Start] A CMakePresets.json is already configured.'));
+        } else {
+            if (!await this.presetsController.selectConfigurePreset(true)) {
+                await this.kitsController.setKitByName(SpecialKits.Unspecified);
+                log.info(localize('cmakepresets.not.created', '[Quick Start] CMakePresets.json not created.'));
+            } else {
+                log.info(localize('cmakepresets.created', '[Quick Start] CMakePresets.json created successfully.'));
```

Comment:
I think we still need to set the `useCMakePresets` context value here as well, so that certain command pallete commands are visible when they should be.

Reply:
Added issue #3734 to solve in a seperate PR